packages/components/src/internal/components/domainproperties/DomainForm.tsx

```diff
@@ -328,6 +332,18 @@ export class DomainFormImpl extends React.PureComponent<IDomainFormInput, IDomai
             }
         }
 
+        // if this domain has any Ontology Lookup field(s), check if we need to update the related field properties based on
+        // the updated domain (i.e. check for any name changes to selected fields)
+        // note: we skip any rowIndexChange which has an newIndex as those are just reorder changes
+        if (
+            rowIndexChange?.newIndex === undefined &&
+            this._fieldAdditionalDetails['ontologyLookupIndices'].length > 0
+        ) {
+            this._fieldAdditionalDetails['ontologyLookupIndices'].forEach(index => {
```

Comment:
Yes, that is what I was getting at when setting this as a class variable. But I agree with you that those side effects are a bad idea. I'll make an update so that the new state variable gets updated on a domain change.